hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -2042,6 +2043,22 @@ public void abort() throws IOException {
 
   @Override
   public boolean needsCompaction() {
+    // For some system compacting, we set selectNow to false, and the files do not
```

Comment:
In general I do not fully understand what is the problem here...

The problem is we schedule too many compactions, because of how we know whether there is a compaction running is through the filesCompacting? This is only true for system compaction? Then why not change the logic of requesting system compaction?

Reply:
The current logic is an improvement introduced by HBASE-8665, which could get better store file selection.
I think that patch is useful, just bring some side effect, and this patch want to fix it.
BTW, skimming that patch maybe helpful to understand.
Thanks a lot.